### PR TITLE
[stdlib] Restore MemoryLayout.*(ofValue:)

### PR DESCRIPTION
**Pull request by xwu**
#### What's in this pull request?

This PR, requested by Dave Abrahams, restores `ofValue` variants of `size`, `stride`, and `alignment`.

I removed the hardcoded fix-its for migrating away from `sizeofValue(_:)` and friends, since the restoration of these counterparts means that a simple `renamed` will do.

I also took the liberty of a little gardening on the doc comments (80-character columns, etc.).

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

A smoke test on macOS does the following:
1. Builds the compiler incrementally.
2. Builds the standard library only for macOS. Simulator standard libraries and
   device standard libraries are not built.
3. lldb is not built.
4. The test and validation-test targets are run only for macOS. The optimized
   version of these tests are not run.

A smoke test on Linux does the following:
1. Builds the compiler incrementally.
2. Builds the standard library incrementally.
3. lldb is built incrementally.
4. The swift test and validation-test targets are run. The optimized version of these
   tests are not run.
5. lldb is tested.

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by tkremenek** [comment]
@swift-ci test

File: test/1_stdlib/Renames.swift [hunk 1]
```diff
@@ -284,9 +284,6 @@ func _MemoryLayout<T>(t: T) {
   _ = sizeof(T.self) // expected-error {{'sizeof' is unavailable: use MemoryLayout<T>.size instead.}} {{7-14=MemoryLayout<}} {{15-21=>.size}} {{none}}
   _ = alignof(T.self) // expected-error {{'alignof' is unavailable: use MemoryLayout<T>.alignment instead.}} {{7-15=MemoryLayout<}} {{16-22=>.alignment}} {{none}}
   _ = strideof(T.self) // expected-error {{'strideof' is unavailable: use MemoryLayout<T>.stride instead.}} {{7-16=MemoryLayout<}} {{17-23=>.stride}} {{none}}
-  _ = sizeofValue(t) // expected-error {{'sizeofValue' is unavailable: use MemoryLayout<T>.size instead.}} {{7-21=MemoryLayout<T>.size}} {{none}}
-  _ = alignofValue(t) // expected-error {{'alignofValue' is unavailable: use MemoryLayout<T>.alignment instead.}} {{7-22=MemoryLayout<T>.alignment}} {{none}}
-  _ = strideofValue(t) // expected-error {{'strideofValue' is unavailable: use MemoryLayout<T>.stride instead.}} {{7-23=MemoryLayout<T>.stride}} {{none}}
```
Comment (gribozavr) [inline test/1_stdlib/Renames.swift]: Could you add tests for the new migration fixits that you added?

**Comment by tkremenek** [comment]
@swift-ci test

File: test/1_stdlib/Renames.swift (hunk 1 above)
Comment (xwu) [inline test/1_stdlib/Renames.swift]: Will do, as soon as I'm in front of a computer this afternoon.

**Comment by dabrahams** [comment]
@swift-ci Please smoke test

**Comment by dabrahams** [comment]
@swift-ci Please smoke test Linux platform

**Comment by xwu** [comment]
@dabrahams Does the Linux failure look related to you?

**Comment by dabrahams** [comment]
@swift-ci Please smoke test Linux platform

**Comment by dabrahams** [comment]
on Fri Aug 05 2016, Xiaodi Wu <notifications-AT-github.com> wrote:

> @dabrahams Does the Linux failure look related to you?

I can imagine one way they might be related, but it seems like a
stretch.  re-testing.

-- 
-Dave

File: stdlib/public/core/MemoryLayout.swift [hunk 2]
```diff
@@ -44,9 +41,29 @@ public enum MemoryLayout<T> {
 }
 
 extension MemoryLayout {
+  /// Returns the contiguous memory footprint of `T`.
+  ///
+  /// Does not include any dynamically-allocated or "remote" storage. In
+  /// particular, `MemoryLayout(ofValue: x).size`, when `x` is a class instance,
```
Comment (benrimmington) [inline stdlib/public/core/MemoryLayout.swift]: Typo in documentation comment: `MemoryLayout(ofValue: x).size` => `MemoryLayout.size(ofValue: x)`

Comment (xwu) [inline stdlib/public/core/MemoryLayout.swift]: Thanks! Nate Cook has just cleaned up the documentation for the existing bits of MemoryLayout quite a bit, so I'll either rebase or submit a follow-up patch and incorporate a correction.

**Comment by xwu** [comment]
@dabrahams Second thoughts on kicking off the review?

**Comment by dabrahams** [comment]
@xwu No, darnit.  I was having technical issues with my email :persevere:.  Should have gone out now.

**Comment by xwu** [comment]
(Rebased.)

**Comment by dabrahams** [comment]
@xwu rebasing doesn't really help unless someone added a merge conflict upstrea.  It just means we need to test again

**Comment by dabrahams** [comment]
@swift-ci Please test

**Comment by xwu** [comment]
I'm aware. Nate Cook did a massive doc comment update that included MemoryLayout; this was not an optional rebase.

**Comment by CodaFi** [comment]
@swift-ci please smoke test OS X platform.

**Pull request merged**